bindings_generator/src/main.rs

```diff
@@ -1,30 +1,29 @@
+use std::path::Path;
+
 fn main() {
-    let out_path = std::env::current_dir().unwrap();
-    let header = out_path.join("wrapper.h").to_str().unwrap().to_string();
-    let out_file = out_path.join("../src/bindings.rs");
+    let crate_root = Path::new(env!("CARGO_MANIFEST_DIR"));
+    let header = crate_root.join("wrapper.h");
+    let out_file = crate_root.join("../src/bindings.rs");
+    let include_dir = crate_root.join("vendor");
 
     bindgen::Builder::default()
-        .header(header)
-        .clang_args(&["-I./vendor/", "-Wno-microsoft-enum-forward-reference"])
+        .header(header.to_str().unwrap())
+        .clang_args(&[
+            format!("-I{}", include_dir.to_str().unwrap()).as_str(),
+            "-Wno-microsoft-enum-forward-reference",
+            "-fretain-comments-from-system-headers",
+        ])
         .dynamic_link_require_all(true)
         .dynamic_library_name("metal_irconverter")
         .layout_tests(false)
         .default_enum_style(bindgen::EnumVariation::Rust {
             non_exhaustive: false,
         })
         .bitfield_enum(".*Flags$")
-        .blocklist_item("__darwin.*")
-        .blocklist_item("__DARWIN.*")
-        .blocklist_item("_DARWIN.*")
-        .blocklist_item("true_")
-        .blocklist_item("false_")
-        .blocklist_item("__bool_true_false_are_defined")
-        .blocklist_item("_opaque_pthread.*")
-        .blocklist_item("__security_init_cookie")
-        .blocklist_item("__security_check_cookie")
-        .blocklist_item("__security_cookie")
-        .blocklist_item("__va_start")
-        .blocklist_item("__report_gsfailure")
+        .generate_comments(true)
+        .allowlist_type("IR\\w+")
+        .allowlist_function("IR\\w+")
+        .anon_fields_prefix("u_")
```

Comment:
Was there also the ability to rename _the type_ from `__bindgen_ty_1` to `_u`?

Reply:
I looked through the functions and I can't find it anywhere, so I'm not sure. The other branch doesn't seem to have this either

Reply:
The other branch very clearly has it: https://github.com/Traverse-Research/saxaboom/compare/main...switch-to-bindgen

Reply:
But we can also cherry-pick features from that branch in a future PR.